lib/fibonacci.rb

```diff
@@ -1,9 +1,22 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n), go through fib for each number one at a time until you reach n
+# Space Complexity - ?  (should be O(n)), storing n fibonacci numbers
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
 def fibonacci(n)
```

Comment:
This works, but you need to maintain the entire array from 1..n in size.  Can you do it without the array?